stl/inc/stacktrace

```diff
@@ -0,0 +1,344 @@
+        return _Result;
+    }
+
+    _NODISCARD string source_file() const {
+        string _Result;
+        __std_stacktrace_source_file(_Address, &_Result, _Stacktrace_string_fill_impl);
+        return _Result;
+    }
+
+    _NODISCARD uint_least32_t source_line() const noexcept /* strengthened */ {
+        return __std_stacktrace_source_line(_Address);
+    }
+
+    _NODISCARD_FRIEND constexpr bool operator==(const stacktrace_entry&, const stacktrace_entry&) noexcept = default;
+
+    _NODISCARD_FRIEND constexpr strong_ordering operator<=>(
+        const stacktrace_entry&, const stacktrace_entry&) noexcept = default;
+
+private:
+    void* _Address = nullptr;
+};
+
+template <class _Alloc>
+class basic_stacktrace {
+private:
+    using _Frames_t = vector<stacktrace_entry, _Alloc>;
+
+public:
+    using value_type             = stacktrace_entry;
+    using const_reference        = const value_type&;
+    using reference              = value_type&;
+    using const_iterator         = typename _Frames_t::const_iterator;
+    using iterator               = const_iterator;
+    using reverse_iterator       = _STD reverse_iterator<iterator>;
+    using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+    using difference_type        = typename _Frames_t::difference_type;
+    using size_type              = typename _Frames_t::size_type;
+    using allocator_type         = _Alloc;
+
+    // __declspec(noinline) to make the same behavior for debug and release.
+    // We force the current function to be always noinline and add its frame to skipped.
+
+    _NODISCARD __declspec(noinline) static basic_stacktrace
+        current(const allocator_type& _Al = allocator_type()) noexcept {
+        basic_stacktrace _Result(_Internal_t{}, _Max_frames, _Al);
+        _Result._Frames.resize(__std_stacktrace_capture(1, static_cast<unsigned long>(_Max_frames),
+            reinterpret_cast<void**>(_Result._Frames.data()), &_Result._Hash));
+        return _Result;
+    }
+
+    _NODISCARD __declspec(noinline) static basic_stacktrace
+        current(size_type _Skip, const allocator_type& _Al = allocator_type()) noexcept {
+        basic_stacktrace _Result(_Internal_t{}, _Max_frames, _Al);
+        _Result._Frames.resize(__std_stacktrace_capture(static_cast<unsigned long>(_Skip + 1),
+            static_cast<unsigned long>(_Max_frames), reinterpret_cast<void**>(_Result._Frames.data()), &_Result._Hash));
+        return _Result;
+    }
+
+    _NODISCARD __declspec(noinline) static basic_stacktrace
+        current(size_type _Skip, size_type _Max_depth, const allocator_type& _Al = allocator_type()) noexcept {
+        basic_stacktrace _Result(_Internal_t{}, _Max_depth, _Al);
+        _Result._Frames.resize(__std_stacktrace_capture(static_cast<unsigned long>(_Skip + 1),
+            static_cast<unsigned long>(_Max_depth), reinterpret_cast<void**>(_Result._Frames.data()), &_Result._Hash));
+        return _Result;
+    }

```

Comment:
The `current()` overloads are `noexcept`, but `basic_stacktrace` is constructing a non-empty `vector`, so allocation failure will terminate. [[stacktrace.basic.ctor]](https://eel.is/c++draft/stacktrace.basic.ctor) says that it should return "an empty `basic_stacktrace` object if the initialization of `frames_` failed".